src/Illuminate/Routing/Middleware/ThrottleRequests.php

```diff
@@ -190,4 +202,21 @@ protected function calculateRemainingAttempts($key, $maxAttempts, $retryAfter =
 
         return 0;
     }
+
+    /**
+     * Load user from request according to $guards.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return mixed
```

Comment:
I believe the return value can be type-hinted as Authenticatable which means that there's no need for method_exists in resolveRequestSignature, the getAuthIdentifier method should always be there.

Reply:
@sisve `$request->user()` does not always return a Authenticatable object, the `model` property in UserProvider not always use Authenticatable trait.

Reply:
I'm building a project that only has API and users are authenticated by OAuth, so User model doesn't have `remember_token` and `password` field, it seems like it's unnecessary to use Authenticatable trait in the User model.

Reply:
I'm talking about the Authenticatable interface, not the trait. You can throw some NotImplementedException in the methods that are irrelevant. Every implementation of the Guard interface is supposed to return an Authenticatable. https://github.com/laravel/framework/blob/5.5/src/Illuminate/Contracts/Auth/Guard.php#L21-L26

Reply:
@sisve that makes sense, thanks